clients/src/main/resources/common/message/StreamsHeartbeatRequest.json

```diff
@@ -64,15 +64,17 @@
           "about": "The cumulative lag for the task" }
       ]
     },
-    { "name": "AssignmentConfigs", "type": "[]AssignmentConfig", "versions": "0+","nullableVersions": "0+", "default": "null",
+    { "name": "AssignmentConfigs", "type": "[]AssignmentConfig", "versions": "0+", "nullableVersions": "0+", "default": "null",
       "about": "Generic array of assignment configuration strings. Null if unchanged since last heartbeat.",
       "fields": [
         { "name": "Key", "type": "string", "versions": "0+",
           "about": "key of the config" },
         { "name": "Value", "type": "string", "versions": "0+",
           "about": "value of the config" }
       ]
-    }
+    },
+    { "name": "ShutdownApplication", "type": "bool", "versions": "0+", "default": false,
+      "about": "Whether all Streams clients in the group should shut down. Can only be defined if memberEpoch = -1" }
```

Comment:
Just because it seems like a useful restriction. Why would you want to shut down the application but remain in the group?

Reply:
I think it would be simpler if all Streams clients shutdown (and leave the group) due to the same event (i.e. receiving the shutdown application error). If the triggering Streams client shuts down when sending the request to shutdown that would be different code path. 
Additionally, in the presence of a network failure the triggering Streams client would request the shutdown and leave the group. The other members of the group would not get the shutdown error. Instead, if the triggering Streams client requests the shutdown and stays in the group until it gets the shutdown error, there would be a bit more control about the application shutting down completely.

Reply:
Good point! Makes sense